pennylane/transforms/decompose.py

```diff
@@ -858,21 +813,37 @@ def _operator_decomposition_gen(  # pylint: disable=too-many-arguments
     elif acceptance_function(op) or max_depth_reached:
         yield op
 
-    elif isinstance(op, (Allocate, Deallocate)):
-        yield op
-
-    elif graph_solution is not None and graph_solution.is_solved_for(op, num_available_work_wires):
+    elif graph_solution and graph_solution.is_solved_for(op, num_available_work_wires):
         op_rule = graph_solution.decomposition(op, num_available_work_wires)
         with queuing.AnnotatedQueue() as decomposed_ops:
             op_rule(*op.parameters, wires=op.wires, **op.hyperparameters)
         decomp = decomposed_ops.queue
-        current_depth += 1
         if num_available_work_wires is not None:
             num_available_work_wires -= op_rule.get_work_wire_spec(**op.resource_params).total
-    else:
+
+    elif enabled_graph() and isinstance(op, GlobalPhase):
+        warnings.warn(
+            "With qml.decomposition.enabled_graph(), GlobalPhase is not assumed to have a "
+            "decomposition. To disable this warning, add `GlobalPhase` to the gate set, or "
+            "assign a decomposition rule to `GlobalPhase` via the `fixed_decomps` keyword "
+            "argument. To make GlobalPhase decompose to nothing, you can import `null_decomp` "
+            "from pennylane.decomposition.decomposition_rule, and assign it to GlobalPhase."
+        )
+        yield op
+
+    elif op.has_decomposition:
         decomp = op.decomposition()
-        current_depth += 1
 
+    else:
+        warnings.warn(
+            f"Operator {op.name} does not define a decomposition to the target gate set and was not found in the "
+            f"target gate set. To remove this warning, add the operator name ({op.name}) or "
+            f"type ({type(op)}) to the gate set.",
+            UserWarning,
+        )
+        yield op
```

Comment:
So right now for an op that has no decomposition neither matrix, it will just stay as what it is instead of like old decomposition erroring out?